File: Codewars/7kyu/7kyu_sum_even_fibonacci_numbers/src/main.go
package main

import "fmt"

func SumEvenFibonacci(limit int) int {
	sum := 2
	for first, second := 1, 2; second <= limit; first, second = second, first + second {
		if 2 < second && 0 == second % 2 {
			sum += second
		}
	}
	return sum
}

func main() {
	fmt.Println(10 == SumEvenFibonacci(8))
	fmt.Println(2 == SumEvenFibonacci(1))
	fmt.Println(2 == SumEvenFibonacci(2))
	fmt.Println(60696 == SumEvenFibonacci(111111))
	fmt.Println(2 == SumEvenFibonacci(5))
	fmt.Println(2 == SumEvenFibonacci(6))
}

File: Codewars/7kyu/7kyu_sum_even_fibonacci_numbers/src/main.py
def SumEvenFibonacci(limit):
    sum_even_fibo = fibo_num = 0
    first, second = 0, 1

    while(fibo_num < limit):
        fibo_num = first + second
        if not fibo_num % 2 and fibo_num <= limit:
            sum_even_fibo = sum_even_fibo + fibo_num
        first, second = second, fibo_num
    
    return sum_even_fibo

print 10 == SumEvenFibonacci(8)
print 0 == SumEvenFibonacci(1)
print 2 == SumEvenFibonacci(2)
print 60696 == SumEvenFibonacci(111111)
print 2 == SumEvenFibonacci(5)
print 2 == SumEvenFibonacci(6)
